blocks/library/button/index.js

```diff
@@ -109,28 +115,31 @@ registerBlockType( 'core/button', {
 							checked={ !! clear }
 							onChange={ toggleClear }
 						/>
-
-						<ColorPalette
-							value={ color }
-							onChange={ ( colorValue ) => setAttributes( { color: colorValue } ) }
-						/>
-						<InspectorControls.TextControl
-							label={ __( 'Hex Color' ) }
-							value={ color || '' }
-							onChange={ ( value ) => setAttributes( { color: value } ) }
-						/>
+						<PanelBody title={ __( 'Button Background Color' ) }>
+							<ColorPalette
+								value={ backgroundColor }
+								onChange={ ( colorValue ) => setAttributes( { backgroundColor: colorValue } ) }
+								withTransparentOption
+							/>
+						</PanelBody>
+						<PanelBody title={ __( 'Button Text Color' ) }>
+							<ColorPalette
+								value={ textColor }
+								onChange={ ( colorValue ) => setAttributes( { textColor: colorValue } ) }
+							/>
+						</PanelBody>
 					</InspectorControls>
 				}
 			</span>,
 		];
 	},
 
 	save( { attributes } ) {
-		const { url, text, title, align, color } = attributes;
+		const { url, text, title, align, backgroundColor, textColor } = attributes;
 
 		return (
-			<div className={ `align${ align }` } style={ { backgroundColor: color } }>
-				<a href={ url } title={ title }>
+			<div className={ `align${ align }` } style={ { backgroundColor: backgroundColor } }>
+				<a href={ url } title={ title } style={ { color: textColor } }>
```

Comment:
I wonder what HTML would this produce if `textColor` is undefined. My thinking is that it should not add any style attribute to the dom node to keep the old blocks "valid"

Reply:
Tested and works as expected

Reply:
I tested the HTML produced and if textColor is null no style is added.